percona/pmm/pmm.go

```diff
@@ -0,0 +1,190 @@
+				Exec: &corev1.ExecAction{
+					Command: []string{
+						"bash",
+						"-c",
+						"pmm-admin inventory remove node --force $(pmm-admin status --json | python -c \"import sys, json; print(json.load(sys.stdin)['pmm_agent_status']['node_id'])\")",
+					},
+				},
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+			{
+				Name:  "PMM_USER",
+				Value: "api_key",
+			},
+			{
+				Name:  "PMM_SERVER",
+				Value: pmmSpec.ServerHost,
+			},
+			{
+				Name:  "PMM_AGENT_SERVER_ADDRESS",
+				Value: pmmSpec.ServerHost,
+			},
+			{
+				Name:  "PMM_AGENT_SERVER_USERNAME",
+				Value: "api_key",
+			},
+			{
+				Name: "PMM_AGENT_SERVER_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: pmmSpec.Secret,
+						},
+						Key: "PMM_API_KEY",
+					},
+				},
+			},
+			{
+				Name:  "CLIENT_PORT_LISTEN",
+				Value: "7777",
+			},
+			{
+				Name:  "CLIENT_PORT_MIN",
+				Value: "30100",
+			},
+			{
+				Name:  "CLIENT_PORT_MAX",
+				Value: "30105",
+			},
+			{
+				Name:  "PMM_AGENT_LISTEN_PORT",
+				Value: "7777",
+			},
+			{
+				Name:  "PMM_AGENT_PORTS_MIN",
+				Value: "30100",
+			},
+			{
+				Name:  "PMM_AGENT_PORTS_MAX",
+				Value: "30105",
+			},
+			{
+				Name:  "PMM_AGENT_CONFIG_FILE",
+				Value: "/usr/local/percona/pmm2/config/pmm-agent.yaml",
+			},
+			{
+				Name:  "PMM_AGENT_SERVER_INSECURE_TLS",
+				Value: "1",
+			},
+			{
+				Name:  "PMM_AGENT_LISTEN_ADDRESS",
+				Value: "0.0.0.0",
+			},
+			{
+				Name: "PMM_AGENT_SETUP_NODE_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.name",
+					},
+				},
+			},
+			{
+				Name:  "PMM_AGENT_SETUP_METRICS_MODE",
+				Value: "push",
+			},
+			{
+				Name:  "PMM_AGENT_SETUP",
+				Value: "1",
+			},
+			{
+				Name:  "PMM_AGENT_SETUP_FORCE",
+				Value: "1",
+			},
+			{
+				Name:  "PMM_AGENT_SETUP_NODE_TYPE",
+				Value: "container",
+			},
+			{
+				Name:  "PMM_AGENT_SIDECAR",
+				Value: "true",
+			},
+			{
+				Name:  "PMM_AGENT_SIDECAR_SLEEP",
+				Value: "5",
+			},
+			{
+				Name:  "PMM_AGENT_PRERUN_SCRIPT",
+				Value: "pmm-admin status --wait=10s; pmm-admin add postgresql --tls-skip-verify --skip-connection-check --metrics-mode=push --username=$(DB_USER) --password='$(DB_PASS)' --service-name=$(PMM_AGENT_SETUP_NODE_NAME) --host=$(POD_NAME) --port=5432 --query-source=pgstatmonitor; pmm-admin annotate --service-name=$(PMM_AGENT_SETUP_NODE_NAME) 'Service restarted'",
```

Comment:
It does not work at all. We do not have any stat in PMM server. 
```
/bin/sh: line 1: DB_USER: command not found
invalid field Username: value '' must not be an empty string
Service with name "cluster1-instance1-tnjw-0" not found.
```

Reply:
https://github.com/percona/percona-postgresql-operator/pull/313/commits/422611dd74c87af724c7bf38f628e6c5286f291a